sparse_strips/vello_api/src/pixmap.rs

```diff
@@ -122,38 +140,60 @@ impl Pixmap {
     }
 
     /// Returns a reference to the underlying data as premultiplied RGBA8.
+    ///
+    /// The pixels are in row-major order. Each pixel consists of four bytes in the order
+    /// `[r, g, b, a]`.
     pub fn data(&self) -> &[u8] {
         &self.buf
     }
 
     /// Returns a mutable reference to the underlying data as premultiplied RGBA8.
+    ///
+    /// The pixels are in row-major order. Each pixel consists of four bytes in the order
+    /// `[r, g, b, a]`.
     pub fn data_mut(&mut self) -> &mut [u8] {
         &mut self.buf
     }
 
     /// Sample a pixel from the pixmap.
+    ///
+    /// The pixel data is [premultiplied RGBA8][PremulRgba8].
     #[inline(always)]
-    pub fn sample(&self, x: u16, y: u16) -> &[u8] {
+    pub fn sample(&self, x: u16, y: u16) -> PremulRgba8 {
         let idx = 4 * (self.width as usize * y as usize + x as usize);
-        &self.buf[idx..][..4]
+
+        // The conversion here is a no-op, as `PremulRgba8`'s fields are in the same memory order
+        // as the pixel data.
+        PremulRgba8::from_u8_array(self.buf[idx..][..4].try_into().unwrap())
     }
 
-    /// Convert from premultiplied to separate alpha.
+    /// Consume the pixmap, returning the data as the underlying [`Vec`] of premultiplied RGBA8
+    /// bytes.
+    ///
+    /// The pixels are in row-major order. Each pixel consists of four bytes in the order
+    /// `[r, g, b, a]`.
+    pub fn take(self) -> Vec<u8> {
+        self.buf
+    }
+
+    /// Consume the pixmap, returning the data as (unpremultiplied) RGBA8 bytes.
     ///
     /// Not fast, but useful for saving to PNG etc.
-    #[expect(
-        clippy::cast_possible_truncation,
-        reason = "cannot overflow in this case"
-    )]
-    pub fn unpremultiply(&mut self) {
+    ///
+    /// The pixels are in row-major order. Each pixel consists of four bytes in the order
+    /// `[r, g, b, a]`.
+    pub fn take_unpremultiplied(mut self) -> Vec<u8> {
         for rgba in self.buf.chunks_exact_mut(4) {
```

Comment:
(Idle thought: I wonder if
```rust
        for [r, g, b, a] in self.buf.chunks_exact_mut(4) {
```
can work. I don't have a good intuition for when the exclusive reference is decomposed in patterns)

Reply:
I think that would work with [`slice::as_chunks_mut`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.as_chunks_mut), which returns a mutable slice of const-sized arrays (`&mut [[T; N]]`). That's nightly-only. [`slice::chunks_exact_mut`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.chunks_exact_mut) is an iterator over slices.

Reply:
Ah, of course. You'd need to use `TryFrom` to make the array of size 4